utils/sites.js

```diff
@@ -282,37 +254,99 @@ const updateParticipantData = async (req, res, authObj) => {
             await siteNotificationsHandler(docData['Connect_ID'], '987563196', docData['827220437'], obj);
         }
 
+        // Flatten dataObj and docData for comparison & validation.
+        const { flattened: flatDataObj, arrayPaths: flatArrayPaths } = flattenObject(dataObj);
+        const { flattened: flatDocData } = flattenObject(docData);
+
+        // Delete primary identifiers from dataObj if they exist. There is no reason to update these through this API.
+        const primaryIdentifiers = ['token', 'pin', 'Connect_ID', 'state.uid'];
+        for (const identifier of primaryIdentifiers) {
+            delete flatDataObj[identifier];
+        }
+
+        // Validate incoming data. flatDocData is only used to check the 'mustExist' property in some rules.
         if(!authObj) {
-            const errors = qc(updatedData, flattened.docData, rules);
+            const errors = flatValidationHandler(flatDataObj, flatDocData, rules, validateUpdateData);
             if(errors.length !== 0) {
                 error = true;
                 responseArray.push({'Invalid Request': {'Token': participantToken, 'Errors': errors}});
                 continue;
             }
         }
 
+        // Check initializeTimestamps and init on match. Currently, only one key exists in initializeTimestamps.
         const { initializeTimestamps } = require('./shared')
+        const keysForTimestampGeneration = Object.keys(initializeTimestamps);
+        for (const key of keysForTimestampGeneration) {
+            if (initializeTimestamps[key] && flatDataObj[key] != null) {
```

Comment:
would `initializeTimestamps[key]` always return true? we are setting that data so I'd think we'd never have to worry about a null value if the key existed?

Reply:
Makes sense and agreed - testing removed version.